### ISSUE TITLE
No license file

### ISSUE THREAD

**Issue by alexandrosk0** [yas-online/UE4-DedicatedServer#11]
Please set this with a reasonable license so it can be used by others legally.
Without a license file, my modifications are technically illegal.

https://help.github.com/en/github/creating-cloning-and-archiving-repositories/licensing-a-repository#choosing-the-right-license
From the above link.
> However, without a license, the default copyright laws apply, meaning that you retain all rights to your source code and no one may reproduce, distribute, or create derivative works from your work.

**Comment by neico** [comment]
I'm not using a license because I'm really not happy with what most licenses offer.

And I am fine with the default copyright until then (as you've quoted)
You can modify the code, and create PR's to be merged in, but I still have the ownership of the code and what to do with it.
And as such only I am allowed to put it onto the Epic Store.

Since I've put the code out there I have given implicit permission to compile and use this plugin wherever you want. (I could have kept it private for example)
And without a proper license you're not even compelled to link back or even credit it, you just can't redistribute it or claim it as your own.

**Comment by alexandrosk0** [comment]
According to what I read, I can't modify the code, only fork and read it.
Making it public doesn't give any permission to use the plugin or modify it, just to read.

I think what you are describing is a Commons Clause license. It's a license that restricts users from selling the software, but allows the rest.
https://commonsclause.com/#faq

https://choosealicense.com/no-permission/
https://opensource.stackexchange.com/questions/1720/what-can-i-assume-if-a-publicly-published-project-has-no-license
In short, the only thing you can safely assume is that you have no rights to do anything at all with this code. In the particular case of GitHub, you can fork the repository and view the code, but nothing more.

**Comment by MarkJGx** [comment]
I agree with @alexandrosk0! Can't use/ship anything with this plugin if it doesn't have a license. 
EDIT: Forgot to mention I appreciate that you released this plugin anyhow!

**Comment by LulzyKing** [comment]
While I absolutely adore the plugin for everything it does @MarkJGx makes a good point. I am concerned about shipping a project with this if there's no license. 
Even if I properly attribute it to you with a massive thank you in my project it still technically breaks laws, as far as I understand.

I 100% understand that licensing is a pain, but is there any legal way to use this plugin @neico?

If not, I still appreciate you releasing this regardless, it gives me hope I can get good enough to write something like this.
